saleor/order/utils.py

```diff
@@ -76,7 +76,7 @@ def get_voucher_discount_assigned_to_order(order: Order):
     return order.discounts.filter(type=OrderDiscountType.VOUCHER).first()
 
 
-def recalculate_order_discounts(order: Order):
+def recalculate_order_discounts(order: Order, with_discount_event: bool = True):
```

Comment:
Are there cases that we want to skip the history log? Flag arguments are no good :(

Reply:
Yes, I am aware, but after 2 hours of trying to find a better solution, I didn't find anything.
For order discount mutations, we call recalculate_order_discounts but we don't want to place the event as each of these mutations has its own specific event. Let me know if you have better way

Reply:
What if the event was created by the caller and the function only returned a boolean, signaling whether anything changed?